src/org/omegat/core/segmentation/SRX.java

```diff
@@ -110,40 +105,105 @@ public SRX copy() {
+                jaxbRule.setBeforebreak(factory.createBeforebreak());
+                jaxbRule.getBeforebreak().setContent(rule.getBeforebreak());                
+                jaxbRule.setAfterbreak(factory.createAfterbreak());                
+                jaxbRule.getAfterbreak().setContent(rule.getAfterbreak());
+            }
+        }
+        
         try {
-            srx.setVersion(CURRENT_VERSION);
-            XMLEncoder xmlenc = new XMLEncoder(new FileOutputStream(outFile));
-            xmlenc.writeObject(srx);
-            xmlenc.close();
-        } catch (IOException ioe) {
+            Marshaller m = SRX_JAXB_CONTEXT.createMarshaller();
+            m.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
+            m.marshal(jaxbObject, new FileOutputStream(outFile));
+        } catch (JAXBException ioe) {
             Log.logErrorRB("CORE_SRX_ERROR_SAVING_SEGMENTATION_CONFIG");
             Log.log(ioe);
-            throw ioe;
+            ioe.printStackTrace();
+            throw new IOException (ioe.getMessage());
         }
     }
-
+    
+    /**
+     * Loads the local segmentation file. Accepts SRX (default) or old CONF format.
+     * In case you use conf format, rules about old version remain valid.
+     **/
+    public static SRX loadSRX(File configDir) {
+        File inFile = null;
+        try { 
+            inFile = new File (configDir, SRX_SENTSEG); 
+            if (inFile.exists()) {
+                return loadSrxFile(inFile.toURI().toURL()); 
+            }
+        } catch (Exception o2) {
+            
+        }
+        
+        // If file was not present or not readable
+        inFile = new File (configDir, CONF_SENTSEG);
+        if (inFile.exists()) { 
+            SRX srx = loadConfFile(inFile); 
+            try {
+                saveToSrx(srx, new File (configDir, SRX_SENTSEG));
+            } catch (Exception o3) {
+                o3.printStackTrace(); // detail why conversion failed, but continue
+            }
+            return srx; 
+        }
+        
+        // Note: as in previous version (4.1) return a new SRX with defaults
+        return SRX.getDefault();
+    }
+    
     /**
      * Loads segmentation rules from an XML file. If there's an error loading a
-     * file, it calls <code>initDefaults</code>.
+     * file, it calls <code>getDefaultFromJar</code>.
      * <p>
      * Since 1.6.0 RC8 it also checks if the version of segmentation rules saved
      * is older than that of the current OmegaT, and tries to merge the two sets
      * of rules.
      */
-    public static SRX loadSRX(File configFile) {
-        if (!configFile.exists()) {
-            return null;
-        }
+    private static SRX loadConfFile (File configFile) {
         SRX res;
         try {
-
             MyExceptionListener myel = new MyExceptionListener();
             XMLDecoder xmldec = new XMLDecoder(new FileInputStream(configFile), null, myel);
```

Comment:
This part was not written by me (Git blame confirmed it to me). I can do the change, but this should not be a reason why to reject the patches.

Reply:
Finally did the change but with a small difference: since XMLDecoder implements AutoClosable and its method close(), according to the doc, "closes the input stream associated with this stream", it is better to use try with XMLDecoder.